libraries/chain/block_state.cpp

```diff
@@ -102,5 +102,26 @@ namespace eosio::chain {
    ,_cached_trxs( std::move(trx_metas) )
    {}
 #endif
+
+   bool block_state::aggregate_vote(const hs_vote_message& vote) {
```

Comment:
Add comment that this is called from net threads.
Also need to make thread safe. Currently we are not setting finalizer_policy, but seems like that can be set at construction so that we do not have to lock it. Likewise the strong_finalizer_digest and weak_finalizer_digest are not mutated after creation. If so we can likely get away with making `pending_qc` thread safe.

Reply:
Please add the comment here as well about called from net threads. We will need to be careful that anything accessed in block_state is safe to access from multiple threads.